cmd/metal-api/internal/metal/provisioning_test.go

```diff
@@ -196,65 +173,57 @@ func TestProvisioning_IncompleteCycles(t *testing.T) {
 		want           string
 	}{
 		{
-			name: "TestProvisioning_IncompleteCycles Test 1",
+			name: "TestFsm_CrashCycles Test 1",
 			eventContainer: ProvisioningEventContainer{
-				Events: SuccessfulEventCycle,
+				Events: SuccessfulCycle,
 			},
 			want: "0",
 		},
 		{
-			name: "TestProvisioning_IncompleteCycles Test 2",
+			name: "TestFsm_CrashCycles Test 2",
 			eventContainer: ProvisioningEventContainer{
-				Events: CrashEventCycle,
+				Events: CrashCycle1,
 			},
 			want: "2",
 		},
 		{
-			name: "TestProvisioning_IncompleteCycles Test 3",
+			name: "TestFsm_CrashCycles Test 3",
 			eventContainer: ProvisioningEventContainer{
-				Events: CycleWithPlannedReboot,
+				Events: PlannedRebootCycle,
 			},
 			want: "0",
 		},
 		{
-			name: "TestProvisioning_IncompleteCycles Test 4",
+			name: "TestFsm_CrashCycles Test 4",
 			eventContainer: ProvisioningEventContainer{
-				Events: CycleWithPlannedRebootAndError,
+				Events: PlannedRebootAndError,
 			},
 			want: "1",
 		},
 		{
-			name: "TestProvisioning_IncompleteCycles Test 5",
+			name: "TestFsm_CrashCycles Test 5",
 			eventContainer: ProvisioningEventContainer{
-				Events: CycleWithPlannedRebootAndImmediateError,
+				Events: CrashCycle2,
 			},
 			want: "1",
 		},
 		{
-			name: "TestProvisioning_IncompleteCycles Test 6",
+			name: "TestFsm_CrashCycles Test 6",
 			eventContainer: ProvisioningEventContainer{
-				Events: CycleWithACrash,
-			},
-			want: "1",
-		},
-		{
-			name: "TestProvisioning_IncompleteCycles Test 7",
-			eventContainer: ProvisioningEventContainer{
-				Events: CycleWithReset,
+				Events: ResetCycle,
 			},
 			want: "0",
-		},
-		{
-			name: "TestProvisioning_IncompleteCycles Test 8",
+		}, {
+			name: "TestFsm_CrashCycles Test 7",
 			eventContainer: ProvisioningEventContainer{
-				Events: SuccessfulEventCycleWithBadHistory,
+				Events: SuccessfulCycleWithBadHistory,
 			},
 			want: "0",
 		},
 		{
-			name: "TestProvisioning_IncompleteCycles Test 9",
+			name: "TestFsm_CrashCycles Test 8",
```

Comment:
General remark, not this line specifically: Just introduce a new test for your code, the `CalculateIncompleteCycles` is not so important for the beginning. In case it starts to fail, please just comment this test.